### ISSUE TITLE
Migrate all tests from Truffle to Hardhat

### ISSUE THREAD

**Issue by udosson** [kumodao/kumo-protocol#189]
Currently, the tests are either written for `Truffle` or `Hardhat`. This has a historical reason. Liquity started with Truffle but then moved to Hardhat. To prevent rewriting all of the Truffle tests, they used the wrapper `nomiclabs/hardhat-truffle5`, which allows tests and scripts written for Truffle to work with Hardhat.

Two reasons why we want to remove Truffle completely and rewrite scripts and tests:
- All tests rely on the same framework, which makes the code cleaner and prevents compatibility issues
- We need to implement the UUPS proxy for our smart contracts and decided to use `@openzeppelin/contracts-upgradeable` and `@openzeppelin/hardhat-upgrades`. Both are not compatible with `nomiclabs/hardhat-truffle5` and require plain hardhat.

The tests are at `packages/contracts/test` and the helpers are at `packages/contracts/utils/`

**Comment by arthurka-o** [comment]
After a few months, I think we're okay with the current situation. The second reason is obsolete after a small trick. There must be more than the first reason to justify such significant refactoring.
@udosson

**Comment by udosson** [comment]
I agree. Let's move it to the backlog and review the task again in 2-3 months
